src/arch/riscv/reg_abi.hh

```diff
@@ -68,7 +70,29 @@ struct Argument<ABI, Arg,
     {
         panic_if(state >= ABI::ArgumentRegs.size(),
                 "Ran out of syscall argument registers.");
-        return bits(tc->getReg(ABI::ArgumentRegs[state++]), 31, 0);
+
+        auto low = ABI::ArgumentRegs[state++];
+        auto high = ABI::ArgumentRegs[state++];
+        return (Arg)ABI::mergeRegs(tc, low, high);
+    }
+};
+
+// This method will be used for Riscv 32 pointers.
+template <typename ABI, typename Arg>
+struct Argument<ABI, Arg,
+    typename std::enable_if_t<
+        std::is_base_of_v<RiscvISA::RegABI32, ABI> &&
+        std::is_same<Arg,GuestAddr>::value &&
+        ABI::template IsWideV<Arg>>>
+{
+    static Arg
+    get(ThreadContext *tc, typename ABI::State &state)
+    {
+        panic_if(state >= ABI::ArgumentRegs.size(),
+                "Ran out of syscall argument registers.");
+
+        auto arg = bits(tc->getReg(ABI::ArgumentRegs[state++]), 31, 0);
+        return *reinterpret_cast<Arg*>(&arg);
```

Comment:
Sorry for the wrong comment. Please ignore latest one.

the type of `time_t` and `off_t` are changed to 64bits [link](https://patchwork.ozlabs.org/project/glibc/patch/4e95f95966d8d7c6a8339160dc62d81c1f6a1bfb.1578824547.git.alistair.francis@wdc.com/). The syscall of [macros/mt](https://github.com/gem5/gem5-resources/tree/stable/src/asmtest/isa/macros/mt) need to be updated